lib/Futures/Utilities.h

```diff
@@ -158,6 +162,35 @@ auto collectAll(Collection&& c) -> decltype(collectAll(std::begin(c), std::end(c
   return collectAll(std::begin(c), std::end(c));
 }
 
+template <class Collection, class Rep, class Period>
+typename Collection::iterator waitForAny(Collection& c,
+                                         std::shared_ptr<std::condition_variable> cv,
+                                         std::chrono::duration<Rep, Period> timeout) {
+  using T = typename std::iterator_traits<typename Collection::iterator>::value_type::value_type;
+
+  auto ready = [](Future<T> const& f) -> bool { return f.isReady(); };
+  auto it = std::find_if(std::begin(c), std::end(c), ready);
+  if (it != std::end(c)) {
+    return it;
+  }
+
+  std::mutex m;
```

Comment:
@jsteemann It's not, really. The mutex is there primarily just as a dummy to satisfy the `condition_variable::wait_for` API. The `condition_variable` itself is passed in to avoid a lifecycle issue where we do a repeated wait. Essentially, we want to be able to wait on the same `condition_variable` over the lifetime of the set of futures, since once you attach a callback to a future, you can't attach another. Do you think I should hide this behind some `futures::WaitState` struct or some such to make it a bit less prone to user-error?